### PR TITLE
feat: add `mamba` spec

### PR DESCRIPTION
**Pull request by Lunchb0ne**
[Mamba](https://github.com/mamba-org/mamba) is a faster alternative for conda written in C++.

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
New Spec for `mamba` 

Most of it is taken from the spec for `conda`

**Additional info:**

### CODE REVIEW

File: src/mamba.ts [hunk 1]
```diff
@@ -0,0 +1,1120 @@
...
+          name: parts[0],
+          description: parts[parts.length - 1],
+          priority: isActive ? 100 : 50,
+          icon: isActive ? "✅" : "🐍",
+        });
+      }
+      return availableEnvs;
+    } catch (e) {
+      return availableEnvs;
+    }
```
Comment (fedeci) [inline src/mamba.ts:40]: Is the trycatch really required?

File: src/mamba.ts [hunk 2]
```diff
@@ -0,0 +1,1120 @@
...
+};
+
+const getInstalledPackages: Fig.Generator = {
+  script: "conda list --json",
+  scriptTimeout: 10000,
+  cache: {
+    strategy: "stale-while-revalidate",
+  },
+  postProcess: function (out) {
+    const parsed: Array<Package> = JSON.parse(out);
```
Comment (fedeci) [inline src/mamba.ts:51]: This should be trycatched

File: src/mamba.ts [hunk 3]
```diff
@@ -0,0 +1,1120 @@
...
+        {
+          name: "--stack",
+          description:
+            "Stack the environment being activated on top of the previous active environment",
+        },
+        {
+          name: "--no-stack",
+          description:
+            "Do not stack the environment. Overrides 'auto_stack' setting",
+        },
```
Comment (fedeci) [inline src/mamba.ts:373]: These are mutually `exclusiveOn` (see https://fig.io/docs/reference/option#exclusiveon)

**Review by fedeci — DISMISSED** [review]
Great one, thanks! I left two small comments to solve and we are ready to merge this.

File: src/mamba.ts (hunk 1 above)
Comment (Lunchb0ne) [reply src/mamba.ts]: Ah, it's not necessary now. Updated!

File: src/mamba.ts (hunk 2 above)
Comment (Lunchb0ne) [reply src/mamba.ts]: Done!

File: src/mamba.ts (hunk 3 above)
Comment (Lunchb0ne) [reply src/mamba.ts]: I updated some options with `exclusiveOn` and `dependsOn`.

Comment (fedeci) [reply src/mamba.ts]: Great, thanks!

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Spec is looking really good! One request - can you add a `name` prop everywhere you see `args: {}`? I know this is just inherited from conda but don't worry about fixing it there

Looks better in https://fig.io/manual because it describes what the argument is for :) no need to add a description to them

**Review by SeparateRecords — APPROVED** [review]
lgtm! Thank you so much for the contribution, nice job on it!

**Pull request merged**